octree_web_viewer/src/state.rs

```diff
@@ -0,0 +1,101 @@
+use crate::backend_error::PointsViewerError;
+use point_viewer::octree;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+/// path information for the octrees
+#[derive(Clone)]
+pub struct OctreeKeyParams {
+    /// Location prefix, including
+    prefix: String,
+    /// Tree ID
+    suffix: String,
+}
+
+impl OctreeKeyParams {
+    pub fn get_octree_address(&self, octree_key: &String) -> Result<String, PointsViewerError> {
+        let mut join_prefix = "";
+        let mut join_suffix = "";
+
+        if !self.prefix.ends_with("/") {
+            join_prefix = "/";
+        }
+        if !self.suffix.starts_with("/") {
```

Comment:
What if the suffix is empty?

Reply:
outdated, using PathBuf s. comment on the next lines